### PR TITLE
implement SocketData for monitoring

### PR DESCRIPTION
**Pull request by jaime-ez**
plus other fixes, specially for uws service

### CODE REVIEW

**Comment by yasserf** [comment]
Thanks!

I published deepstream.io-types, should we update that in package.json?

**Comment by yasserf** [comment]
Also what version of node are you running? The error messages from node seem to differ between the two

File: src/services/http/uws/uws-http.ts [hunk 1]
```diff
@@ -224,6 +265,84 @@ export class UWSHTTP extends DeepstreamPlugin implements DeepstreamHTTPService {
...
+      this.terminateResponse(response, HTTPStatus.FORBIDDEN, 'Forbidden Host.')
+      return false
+    }
+
+    if (this.origins.indexOf(requestOriginUrl) === -1) {
+      if (!requestOriginUrl) {
+        this.terminateResponse(
+          response,
+          HTTPStatus.FORBIDDEN,
+          'CORS is configured for this this. All requests must set a valid "Origin" header.'
```
Comment (yasserf) [inline src/services/http/uws/uws-http.ts:282]: double this

**Comment by jaime-ez** [comment]
> Also what version of node are you running? The error messages from node seem to differ between the two

v10.15.3 on Ubuntu 18.04.4...yes I suspected maybe the test errors where env related

**Comment by yasserf** [comment]
awesome this looks great thank you!

**Comment by yasserf** [comment]
If you can raise a changelog I'm happy to release ASAP

**Comment by yasserf** [comment]
Thanks a million for all the work across repos done for this. Very nicely done!

**Comment by jaime-ez** [comment]
glad to contribute :)

**Pull request merged**